Neos.Utility.Arrays/Classes/PositionalArraySorter.php

```diff
@@ -113,25 +113,19 @@ public function toArray(): array
     public function getSortedKeys(): array
     {
         $arrayKeysWithPosition = $this->collectArrayKeysAndPositions();
+        $existingKeys = $arrayKeysWithPosition;
 
         $this->extractMiddleKeys($arrayKeysWithPosition);
         $this->extractStartKeys($arrayKeysWithPosition);
         $this->extractEndKeys($arrayKeysWithPosition);
-        $this->extractBeforeKeys($arrayKeysWithPosition);
-        $this->extractAfterKeys($arrayKeysWithPosition);
+        $this->extractBeforeKeys($arrayKeysWithPosition, $existingKeys);
+        $this->extractAfterKeys($arrayKeysWithPosition, $existingKeys);
 
         foreach ($arrayKeysWithPosition as $unresolvedKey => $unresolvedPosition) {
             throw new Exception\InvalidPositionException(sprintf('The positional string "%s" (defined for key "%s") is not supported.', $unresolvedPosition, $unresolvedKey), 1379429920);
         }
 
-        $sortedKeysMap = $this->generateSortedKeysMap();
-
-        $sortedKeys = [];
-        array_walk_recursive($sortedKeysMap, function ($value) use (&$sortedKeys) {
-            $sortedKeys[] = $value;
-        });
-
-        return $sortedKeys;
+        return $this->generateSortedKeysMap();
```

Comment:
Do you know why the previous code is not needed any longer?

Reply:
Ah, the `array_merge([], ...$sortedKeysMap)` below, right?

But now the result will be an associative array, right?
If so, would it make sense to do
```suggestion
        return array_values($this->generateSortedKeysMap());
```
instead?

Reply:
Nevermind. The result is actually a numeric array and a unit test even covers that case